src/twisted/internet/test/test_base.py

```diff
@@ -183,6 +183,7 @@ def setUp(self):
         Create two L{DelayedCall} instanced scheduled to run at different
         times.
         """
+        self.patch(DelayedCall, 'debug', True)
```

Comment:
So… why change all tests to set `debug` on (except `test_str` below)?

The result is that now we are *not* testing everything with `debug=False`, which is the normal config.  It would make more sense to me to subclass this test suite and set `debug=True` in that subclass, so we are testing in both configs, if that's the approach.

But simpler still would be to add tests for `__str__` and `cancel` and `__init__` to test how they behave with `debug=True` and when `self._str` is set.  That seems more unit-test-y.